### ISSUE TITLE
Remove class_name in favor of class

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#1208]
**Describe the project you are working on:**
A spaceship game

**Describe the problem or limitation you are having in your project:**
Not related to the issue I just think the way script work can be less confusing

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
I think class_name should be removed in favor of a more simplistic declaration
```
#first class is used by script
class MyCustomClass extends Node2D: #class name is required to to identify object
    export(int) var index

    class MyCustomSubClass:
         pass

#second one is serialized as object and added but won't be used directly by nodes
class MyCustomClass2 #no extends
    export(int) var index2

    class MyCustomSubClass2:
         pass
``` 

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
basically GDScripts would be classes and could contain sub classes.
this is how it should work since a script is a object and a object is a class

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
this would be used everytime you make a class and not since it would be a syntax change

**Is there a reason why this should be core and not an add-on in the asset library?:**
I think it would more clearly define how godot is a oop engine and how every object is a class

**Comment by Calinou** [comment]
Duplicate of #209. (The proposed syntax is different, but the underlying idea is the same.)

Please continue the discussion there.
